device/common/include/traccc/seeding/device/impl/form_spacepoints.ipp

```diff
@@ -28,17 +27,18 @@ inline void form_spacepoints(
         return;
     }
 
+    // Create the tracking geometry
+    detector_t det(det_view);
+
     // Get device copy of input/output parameters
     const measurement_collection_types::const_device measurements(
         measurements_view);
     assert(measurements.size() == measurement_count);
-    const silicon_detector_description::const_device det_descr(det_descr_view);
     spacepoint_collection_types::device spacepoints(spacepoints_view);
-    assert(spacepoints.size() == measurement_count);
 
     // Fill the spacepoint using the common function.
-    details::fill_spacepoint(spacepoints.at(globalIndex),
-                             measurements.at(globalIndex), det_descr);
+    spacepoints.push_back(
+        details::create_spacepoint(det, measurements.at(globalIndex)));
```

Comment:
It's a good idea to make the spacepoint container resizable. Since we don't know how many spacepoints we'll actually create from all of the measurements. But for this, we need to:
  - Make sure that all algorithms would create resizable spacepoint buffers. (Make use of `vecmem::data::buffer_type::resizable`!)
  - Make sure in this function that we would skip the non-2D measurements. Right now I believe the code will call the `create_spacepoint(...)` function for strips as well. :thinking:

Reply:
Sure. A couple of tests (`test_spacepoint_formation` for CPU/CUDA/SYCL) already make sure that they fail without the `resizable` tag.

I decided to drop the 2D condition in this PR and add it into the next one

Reply:
It's a bit of a pity to switch to resizable buffers, for no benefit right now. :thinking: Since filling resizable buffers is always slower. But if we don't know their size, that's of course what we need to accept. :thinking:

In any case: You forgot to make the buffer resizable in the Alpaka algorithm. At least that needs to be fixed!

Reply:
Right - fixed now